src/AppBuild/Bundle/UserBundle/Controller/SecurityController.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace AppBuild\Bundle\UserBundle\Controller;
+
+use AppBuild\Bundle\UserBundle\Entity\User;
+use AppBuild\Bundle\UserBundle\Form\Type\UserType;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+
+class SecurityController extends Controller
+{
+    public function loginAction(Request $request)
+    {
+        $authenticationUtils = $this->get('security.authentication_utils');
+
+        // get the login error if there is one
+        $error = $authenticationUtils->getLastAuthenticationError();
+
+        // last username entered by the user
+        $lastUsername = $authenticationUtils->getLastUsername();
+
+        return $this->render(
+            'AppBuildUserBundle:Security:login.html.twig',
+            array(
+                // last username entered by the user
+                'last_username' => $lastUsername,
+                'error' => $error,
+            )
+        );
+    }
+
+    public function registerAction(Request $request)
+    {
+        $user = new User();
+        $form = $this->createForm(new UserType(), $user);
+
+        $form->handleRequest($request);
+        if ($form->isValid() && $form->isSubmitted()) {
+            $password = $this->get('security.password_encoder')
+                ->encodePassword($user, $user->getPassword());
+            $user->setPassword($password);
+            $em = $this->getDoctrine()->getManager();
+            $em->persist($user);
+            $em->flush();
+
+            return $this->redirectToRoute('appbuild_user_login');
+        }
+
+        return $this->render(
+            'AppBuildUserBundle:Security:register.html.twig',
+            array('form' => $form->createView())
+        );
+    }
+
+    public function editUserAction(Request $request)
+    {
+        if (!$this->get('security.authorization_checker')->isGranted('IS_AUTHENTICATED_FULLY')) {
+            throw $this->createAccessDeniedException();
+        }
+
+        $user = $this->getUser();
+        $form = $this->createForm(new UserType(), $user);
```

Comment:
pour info (osef si on change pas), le créateur du composant Form dit que ce n'est pas une bonne pratique de faire un `new MonFormType()` (parce que c'est pas "dependency injection friendly" sans doute), à la place il vaut mieux passer soit `$this->container->get('mon.form.type.service')`, soit simplement `mon_form` (le nom du form défini comme service et tagué comme form)